CRM/ACL/BAO/ACL.php

```diff
@@ -730,7 +730,7 @@ public static function check($str, $contactID) {
    */
   public static function buildAcl($contactId) {
     // If there is no $contactId passed return empty ACL join and where clause
-    if (empty($contactId)) {
+    if (empty($contactId) || CRM_Core_Permission::check('administer CiviCRM')) {
```

Comment:
It is definitely possible to have administer civicrm permission and NOT be allowed to view all contacts - so this line looks very worrying to me